libmultilabel/model.py

```diff
@@ -60,34 +61,48 @@ def training_step(self, batch, batch_idx):
 
     def validation_step(self, batch, batch_idx):
         loss, pred_logits = self.shared_step(batch)
+
         return {'loss': loss.item(),
                 'pred_scores': torch.sigmoid(pred_logits).detach().cpu().numpy(),
                 'target': batch['label'].detach().cpu().numpy()}
 
+    def validation_step_end(self, batch_parts):
+        pred_scores = np.vstack(batch_parts['pred_scores'])
+        target = np.vstack(batch_parts['target'])
+        self.eval_metric.add_values(target, pred_scores)
+
     def validation_epoch_end(self, step_outputs):
-        eval_metric = self.evaluate(step_outputs, 'val')
-        return eval_metric
+        return self.evaluate(step_outputs, 'val')
 
     def test_step(self, batch, batch_idx):
         return self.validation_step(batch, batch_idx)
 
+    def test_step_end(self, batch_parts):
+        self.validation_step_end(batch_parts)
```

Comment:
not sure if this is a good practice? Should have a separate fun called by both eval and test?
If we really want to do this, I think we should have val to call test (as val is an optional thing in ML)

Reply:
Create separate functions as shown in PyTorch Lightning's document.
https://pytorch-lightning.readthedocs.io/en/latest/common/lightning_module.html#inference-in-production